src/cli/project/export/conda_explicit_spec.rs

```diff
@@ -0,0 +1,220 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use clap::Parser;
+
+use crate::cli::cli_config::PrefixUpdateConfig;
+use crate::cli::LockFileUsageArgs;
+use crate::lock_file::UpdateLockFileOptions;
+use crate::Project;
+use rattler_conda_types::{ExplicitEnvironmentEntry, ExplicitEnvironmentSpec, Platform};
+use rattler_lock::{CondaPackage, Package, PackageHashes, PypiPackage, PypiPackageData, UrlOrPath};
+
+#[derive(Debug, Parser)]
+#[clap(arg_required_else_help = false)]
+pub struct Args {
+    /// Environment to render
+    #[arg(short, long)]
+    environment: Option<String>,
+
+    /// The platform to render. Defaults to the current platform.
+    #[arg(long)]
+    pub platform: Option<Platform>,
```

Comment:
A high-value property of `conda-lock` is being able to get _all_ the locks for a single env. For `pixi`, I'd probably _also_ want multiple environments, by default all. This explosion is handled with the `--filename-template`, where there are a few known tokens available. Also `constructor` really prefers `.txt` as an extension, but who knows.

Putting these together:

```toml
[tasks.generate-flat-locks]
cmd = """
pixi project export conda_explicit_spec \
  --environment "*"
  --filename-template "locks/{enviroment}_{platform}.{suffix}"
  --conda-suffix "conda.lock.txt"
  --pip-suffix "requirements.txt"
"""
inputs = ["pixi.lock"]
outputs = ["locks"]
```

Might yield:

```
build_linux-64.conda.lock.txt
build_linux-64.pip-requirements.txt
build_win-64.conda.lock.txt
build_win-64.pip-requirements.txt
test_linux-64.conda.lock.txt
test_linux-64.pip-requirements.txt
test_win-64.conda.lock.txt
test_win-64.pip-requirements.txt
```

Or:


```toml
cmd = """
pixi project export conda_explicit_spec \
  --environment "*"
  --filename-template "locks/{enviroment}/{platform}/{suffix}"
  --conda-suffix "conda.lock.txt"
  --pip-suffix "requirements.txt"
"""
```

Would generate something more like:

```
build/
  linux-64/
    conda.lock.txt
    pip-requirements.txt
  win-64/
    conda.lock.txt
    pip-requirements.txt
test/
  linux-64/
    conda.lock.txt
    pip-requirements.txt
  win-64/
    conda.lock.txt
    pip-requirements.txt
```

Reply:
I'm happy to add a way to dump all env/platform combinations if that's what the pixi team thinks is the way to go. I'm not sure about `--filename-template`. I think that's non-trivial to do in rust, or at least I'm not sure how best to implement that. Indicatif does something similar with progress bar templating, so that could give some ideas. Would it be sufficient to have an option to dump all, but the file naming convention is not flexible? Currently it's just `conda-{platform}-{env}.lock` and `requirements-{platform}-{env}.txt`.

Reply:
I love that. Would love to see that either in this PR or a followup!

Reply:
Sure, the minimal thing for e.g. `miniforge`-style repos using `constructor` would be: 
- mostly-fixed (but sufficiently verbose) file names
  - a `.txt` ending for the conda-lock so that `constructor` knows what to do with it _without_ invoking the solver again
- ideally an `--output-folder` (which it would ensure exists)

I would _still want to check these in_, as they are pretty much optimal inputs to `inputs/depends-on` and, by extension, CI caches, while `pixi.lock` is generally too-broad (e.g. don't rebuild the whole product just because you changed the test procedure).

Reply:
@bollwyvl Ok, so from what you're saying as well as the suggestions in https://github.com/prefix-dev/pixi/pull/1873#issuecomment-2304118162, we want to create an output directory that will contain the exported files. The linked comment just has it as a positional argument, while you're suggesting a flag. I'm not sure which would be preferred. Other questions/comments:
- Do we need to handle warnings or errors for overwriting existing outputs, or just let it overwrite?
- Makes sense to use the `.txt` suffix since that's in the CEP standard. I was just basing it on the naming convention from the explicit render in `conda-lock` which used `.lock`

Reply:
> positional argument, while you're suggesting a flag

Positional, but defaults to `.` seems reasonable as...

> warnings or errors for overwriting existing outputs

...while I haven't kicked the tires, I imagine the runtime of this is probably so small it won't matter. 

> naming convention from the explicit render in conda-lock

`conda-lock` lets a user emit whatever they want, and doesn't generally talk about a specific named environment, so _always_ needed something in anything but the most trivial case. Or, put differently, it never bothered me enough to PR to change it.